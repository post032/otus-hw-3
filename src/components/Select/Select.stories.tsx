import React from "react";
import { withKnobs, array } from "@storybook/addon-knobs";
import { Select } from "./Select";

export default {
  title: "Input",
  decorators: [withKnobs],
};

export const select = () => [
  <Select key="jsx" label="Symbol" options={array("options", ["X", "Y"])} />,
];
