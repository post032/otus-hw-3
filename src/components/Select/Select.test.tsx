import React from "react";
import renderer from "react-test-renderer";
import { Select } from "./Select";

describe("Select", () => {
  it("renders Select with two options", () => {
    expect(
      renderer.create(<Select label="Symbol" options={["X", "Y"]} />).toJSON()
    ).toMatchSnapshot();
  });
});
