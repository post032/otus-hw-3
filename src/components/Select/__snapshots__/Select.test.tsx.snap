// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders Select with two options 1`] = `
<label>
  Symbol
  <select>
    <option>
      X
    </option>
    <option>
      Y
    </option>
  </select>
</label>
`;
